src/jit/codegenxarch.cpp

```diff
@@ -6252,6 +6252,51 @@ void CodeGen::genCallInstruction(GenTreePtr node)
 
     if (target != nullptr)
     {
+#ifdef _TARGET_X86_
+        if (((call->gtFlags & GTF_CALL_VIRT_KIND_MASK) == GTF_CALL_VIRT_STUB) && (call->gtCallType == CT_INDIRECT))
+        {
+            // On x86, we need to generate a very specific pattern for indirect VSD calls:
+            //
+            //    3-byte nop
+            //    call dword ptr [eax]
+            //
+            // Where EAX is also used as an argument to the stub dispatch helper. Make
+            // sure that the call target address is computed into EAX in this case.
+
+            assert(target->isContainedIndir());
+
+            // Disable random NOP emission
+            getEmitter()->emitDisableRandomNops();
+
+            GenTreeIndir* indir = target->AsIndir();
+            assert(indir->Addr() == indir->Base());
+            assert(indir->HasBase());
+            assert(!indir->HasIndex());
+            assert(indir->Scale() == 1);
+            assert(indir->Offset() == 0);
```

Comment:
I don't think we need to assert these.  All we need is ctrlExpr = GT_IND(some Tree node T) and T got computed to EAX.